### PR TITLE
calcite-chip - don't show X unless new dismissible property is true

### LINKED ISSUE #490: provide a way to not show the dismiss X in chips
There are places where we want to use the chips in a "read only" view, so we don't want to show the "X" b/c we don't want users to be able to dismiss it.

To me a `dismissible` property would make sense, but for the current behavior that would have to default to true, which means you couldn't just use the presence of the attribute to indicate true. Defaulting to false would be a breaking change. I'm having a hard time coming up w/ an opposite term other than `readOnly`.

If maintainers agree w/ the idea and could weigh in on the API I'd be glad to take a crack at a PR.

### PR DESCRIPTION
**Pull request by tomwayson**
resolves #490 

I think I got everything (storybook, tests, etc), but please take a close look as this is my first PR to this repo.

image

### CODE REVIEW

**Comment by tomwayson** [comment]
cc @mjuniper

File: src/components/calcite-notice/calcite-notice.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ export class CalciteNotice {
   @Prop({ mutable: true, reflect: true }) width: "auto" | "half" | "full" =
     "auto";
 
-  /** Select theme (light or dark) */
+  /** Optionally show a button the user can click to dismiss the notice */
```
Comment (macandcheese) [inline src/components/calcite-notice/calcite-notice.tsx:65]: ✨

**Comment by tomwayson** [comment]
Thanks @macandcheese, I made the changes you suggested.

image

I also did a search for any other place the component was used and found something in the demos folder. I'm not sure how those are run, but I added `dismissible` to the one w/ an event listener there too.

**Comment by macandcheese** [comment]
Looks great, thanks for the add!

**Pull request merged**